### DISCUSSION TITLE
Store workflow in database Using Workflow Designer

### DISCUSSION THREAD

**Discussion by makattar** [elsa-workflows/elsa-core#1236] [Q&A] [answered]
I have embedded Elsa Designer in Angular .When i start creating workflow i get workflow activities in json format .

activities : [{ "id": "bc994f79-83da-47f6-8d5a-0891171b3664", "type": "ReceiveHttpRequest", "left": 74, "top": 178, "state": { "path": "/gets", "method": "GET", "readContent": null, "name": "", "title": "", "description": "" }, "blocking": false, "executed": false, "faulted": false }]

This format is different from workflow definition database structure format.
I have end point  which will store workflows.How should is store this workflow in elsa 2.0.

**Comment by sfmskywalker** [comment, marked as answer]
If you use the Elsa 2 designer, then it will already provide the correct format for the workflow definition being edited. You should be able to take that model as-is and post it to Elsa 2’s API endpoint from your own API using an HTTP client. If your API is written in .NET, you can use the Elsa.Client package for that.

**Reply by makattar** [reply]
Thanks for your quick response sir.
I am using npm package of elsa designer (https://github.com/elsa-workflows/elsa-designer) at angular site.And using Elsa 2.0 at .net core backend side.

Also referred how to integrate elsa designer with angular  from https://codesandbox.io/s/angular-elsa-designer-dkb4t?file=/src/assets/workflow.json:0-36835

Now I have array of activities , how should i store them at backend site as workflows?

**Reply by sfmskywalker** [reply]
I see now. The elsa designer you're using is the V1 designer, and won't work with Elsa 2 - which has a new set of components found [here](https://github.com/elsa-workflows/elsa-core/tree/master/src/designer/elsa-workflows-studio).

If you embed the V2 designer, you won't have to store anything yourself - the designer already communicates with the Elsa Server API for you.

**Reply by makattar** [reply]
Thank you sir.
